core/jobs/transforms/firebase_transforms.py

```
@@ -0,0 +1,164 @@
+        used_count = 0
+        fail_count = 0
+        error_messages = []
+
+        while batch := list(itertools.islice(input_iter, self.BATCH_LIMIT)):
+            try:
+                output: OutputType = self.handle_batched_items(batch)
+            except (ValueError, firebase_exceptions.FirebaseError) as e:
+                fail_count += len(batch)
+                error_messages.append(
+                    f'at slice=[{used_count}:{used_count + len(batch)}]: {e}'
+                )
```

Ruff (0.15.15)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)